assume/units/building.py

```diff
@@ -0,0 +1,403 @@
+            objective_value = instance.obj_rule()
+            logger.debug(f"The value of the objective function is {objective_value}.")
+        elif results.solver.termination_condition == TerminationCondition.infeasible:
+            logger.debug("The model is infeasible.")
+        else:
+            logger.debug("Solver Status: ", results.solver.status)
+            logger.debug(
+                "Termination Condition: ", results.solver.termination_condition
+            )
+        # Total power input series
+        temp_1 = instance.total_power_input.get_values()
+        self.opt_power_requirement = pd.Series(data=temp_1)
+        self.opt_power_requirement.index = self.index
+
+        # Variable cost series
+        temp_2 = instance.variable_cost.get_values()
+        self.variable_cost_series = pd.Series(data=temp_2)
+        self.variable_cost_series.index = self.index
+
+    def set_dispatch_plan(
+        self,
+        marketconfig: MarketConfig,
+        orderbook: Orderbook,
+    ) -> None:
+        """
+        Adds the dispatch plan from the current market result to the total dispatch plan and calculates the cashflow.
+
+        Args:
+            marketconfig (MarketConfig): The market configuration.
+            orderbook (Orderbook): The orderbook.
+        """
+        products_index = get_products_index(orderbook)
+
+        for order in orderbook:
+            start = order["start_time"]
+            end = order["end_time"]
+            end_excl = end - self.index.freq
+            if isinstance(order["accepted_volume"], dict):
+                self.outputs["energy"].loc[start:end_excl] += [
+                    order["accepted_volume"][key]
+                    for key in order["accepted_volume"].keys()
+                ]
+            else:
+                self.outputs["energy"].loc[start:end_excl] += order["accepted_volume"]
+
+        self.calculate_cashflow("energy", orderbook)
+
+        for start in products_index:
+            current_power = self.outputs["energy"][start]
+            self.outputs["energy"][start] = current_power
+
+        self.bidding_strategies[marketconfig.market_id].calculate_reward(
+            unit=self,
+            marketconfig=marketconfig,
+            orderbook=orderbook,
+        )
+
+    def calculate_marginal_cost(self, start: pd.Timestamp, power: float) -> float:
+        """
+        Calculate the marginal cost of the unit based on the provided time and power.
+
+        Args:
+            start (pandas.Timestamp): The start time of the dispatch.
+            power (float): The power output of the unit.
+
+        Returns:
+            float: the marginal cost of the unit for the given power.
+        """
+        # Initialize marginal cost
+        marginal_cost = 0
+
+        if self.opt_power_requirement[start] > 0:
+            marginal_cost = (
+                self.variable_cost_series[start] / self.opt_power_requirement[start]
+            )
+        else:
```

Comment:
Since `marginal_cost` got already initialized with 0 the `else` branch is not needed